### PR TITLE
test(qa): verify happy path for partition leave and join

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Related to #14905

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/PartitionJoinTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  void canJoinPartition(final Scenario scenario) {
+    // given
+    try (final var cluster = setupCluster(scenario.initialClusterState())) {
+      // when
+      final var response = runOperation(cluster, scenario.operation());
+      // then
+      assertChangeIsPlanned(response);
+      Awaitility.await("Requested change is completed in time")
+          .untilAsserted(() -> assertChangeIsCompleted(cluster, response));
+      assertChangeIsApplied(cluster, response);
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/PartitionJoinTest.java:34]: :wrench: Here we should also verify if the final `ClusterTopology` has the broker with newly joined partition.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/Utils.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  public static void assertChangeIsApplied(
+      final TestCluster cluster, final PostOperationResponse response) {
+    final var actuator = ClusterActuator.of(cluster.availableGateway());
+    final var expectedTopology = response.getExpectedTopology();
+    final var currentTopology = actuator.getTopology().getBrokers();
+    assertThat(currentTopology)
+        .usingRecursiveComparison()
+        .ignoringFields("lastUpdatedAt")
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/Utils.java:34]: :question: Here also we can use `.ignoringFieldsOfTypes(OffsetDateTime.class), right?

**Review by deepthidevaki — APPROVED** [review]
:rocket: Great to see happy paths working. :crossed_fingers: until more test cases are added.

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**